src/fromager/commands/build.py

```diff
@@ -137,3 +153,22 @@ def _build(
     )
 
     return wheel_filename
+
+
+def _download_built_wheel(
+    wkctx: context.WorkContext, dist_name: str, dist_version: Version
+) -> pathlib.Path | None:
+    req = Requirement(f"{dist_name}=={dist_version}")
+
+    try:
+        logger.info(f"{req.name}: checking if {req} was already built")
+        wheel_filename, _, _ = sdist.download_wheel(
+            wkctx,
+            req,
+            wkctx.wheels_build,
+            [wkctx.wheel_server_url],
```

Comment:
I think we're going to need a different CLI flag for this case.

IIRC, when `--wheel-server-url` is set, then `start_wheel_server()` doesn't start a server. But we need the local server for most things because we have to assume we're building something that doesn't exist on the remote server. So for this case we're going to need to have 2 different wheel server URLs, for different purposes.

Reply:
Right now if we just pass a remote server to `--wheel-server-url` would it still build the package irrespective of whether it is there on the server or not?

Reply:
Yes, today it always builds everything. The `--wheel-server-url` is used to find the build requirements. So if you're building a package that needs `setuptools` to make a wheel, fromager tells pip to download setuptools from whatever wheel server it knows, either the built-in one or the one provided as an argument on the command line.

Reply:
Right now for build sequence, we go through each item in the build order file (irrespective of the type of requirement it is), download the sources, prepare the sources and then build the wheel. The download and prepare source part should be applicable to something like `setuptools` as well. But when we come to building the actual wheel, will pip simply download `setuptools` from the wheel server instead of building it from the prepared sources because of this option: https://github.com/python-wheel-build/fromager/blob/main/src/fromager/wheels.py#L157 ?

Reply:
Yes, that's right. When the build environment is set up, it [uses the context to get the wheel server](https://github.com/python-wheel-build/fromager/blob/main/src/fromager/wheels.py#L63) and [that code](https://github.com/python-wheel-build/fromager/blob/main/src/fromager/context.py#L68) uses the wheel server passed on the CLI or the one started by fromager if no value is given on the CLI.

Reply:
The code you linked on line 157 should probably be updated to use the same approach as is used in creating the build environment.

Reply:
Doesn't look like we actually every execute [this command](https://github.com/python-wheel-build/fromager/blob/main/src/fromager/wheels.py#L63) for build sequence since we initialize the build environment as [BuildEnvironment(wkctx, source_root_dir.parent, None)](https://github.com/python-wheel-build/fromager/blob/main/src/fromager/commands/build.py#L111) and pass `build_requirements` as None. So in `_create_env` we return before we can execute pip install: https://github.com/python-wheel-build/fromager/blob/main/src/fromager/wheels.py#L51

Reply:
The prepare_build_environment() call doesn't return the BuildEnvironment, so we have to re-instantiate it. We know it exists, so we just pass None as the requirements and it reuses the existing environment that was created [over here](https://github.com/python-wheel-build/fromager/blob/main/src/fromager/sdist.py#L468).

The APIs of those functions should probably be changed so that prepare_build_environment returns the environment it creates, but I'd have to look at what other callers of that function do with the existing return value.

Reply:
I was confused, we don't need a separate option because building only builds 1 thing at a time and uses the wheel server for its dependencies anyway.